XenModel/Actions/Network/NetworkingActionHelpers.cs

```diff
@@ -269,6 +269,45 @@ private static void ReconfigureManagement_(AsyncAction action, PIF pif, int hi)
             action.Description = string.Format(Messages.ACTION_CHANGE_NETWORKING_MANAGEMENT_RECONFIGURED, pif.Name);
         }
 
+        internal static void PoolReconfigureManagement(AsyncAction action, PIF up_pif, PIF down_pif, int hi)
+       {
+           System.Diagnostics.Trace.Assert(down_pif.host.opaque_ref == up_pif.host.opaque_ref);
+
+           int lo = action.PercentComplete;
+           int inc = (hi - lo) / 3;
+           lo += inc;
+           PoolManagementReconfigure_( action, up_pif, lo);
+
+           /* Pool recover slaves spwans two tasks dbsync (update_env) and server_init on each slaves. 
+            * Only after their completion master will be able to execute reconfigure_IPs. 
+            * Hence, we allow some setteling time for them to complete.
+            */
+           Thread.Sleep(10000);

```

Comment:
@stephen-turner , @sharad analyzed it and these are his thoughts  `Pool.management_reconfigure` calls `pool.recover_slaves` this call purposely creates a new task `emergency_reset_master` check here
https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi_pool.ml#L816
Since xapi gets restarted on slaves as part of `emergency_reset_master` we can't make sure task `pool.management_reconfigure` will get finished only when slaves are able to contact with master.